Lib/test/test_functools.py

```diff
@@ -403,6 +403,33 @@ def test_attributes_unwritable(self):
         else:
             self.fail('partial object allowed __dict__ to be deleted')
 
+    def test_manually_adding_non_string_keyword(self):
+        p = self.partial(capture)
+        # Adding a non-string/unicode keyword to partial kwargs
+        p.keywords[1] = 10
+        self.assertIn('1=10', repr(p))
```

Comment:
This test is fragile. Since `1=10` is not valid Python syntax, more comprehensive implementation can return `partial(<foo>, **{1: 10})`.

I would test that the repr() of the partial object contains either str() or repr() of the key and the repr() of the value.
```python
p.keywords[123] = 'value'
r = repr(p)
self.assertIn('123', r)
self.assertIn("'value'", r)
```